lib/gitpassword.sh

```diff
@@ -1,41 +1,71 @@
-#!/bin/bash
+#!/bin/bash -eu
 
-if [ "$1" != "get" ]; then
-	echo "Usage: $0 get"
-	exit 1
-fi
-
-baseDir() {
-	# see https://stackoverflow.com/questions/59895
-	SOURCE=${BASH_SOURCE[0]}
-	while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
-		DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
-		SOURCE=$(readlink "$SOURCE")
-		[[ $SOURCE != /* ]] && SOURCE=$DIR/$SOURCE # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
-        done
-        DIR=$( cd -P "$( dirname "$SOURCE" )/.." >/dev/null 2>&1 && pwd )
-        echo $DIR
-}
+#echo "Called: $@" >> /tmp/credhelper
 
-BASE=$(baseDir)
-cd $BASE
+SECRETS_FILE=/var/cache/bridgehead/secrets/gitetcbridgehead
 
-source lib/functions.sh
+# Called from bridgehead, so shift 2 times
+shift
+shift
 
-assertVarsNotEmpty SITE_ID || fail_and_report 1 "gitpassword.sh failed: SITE_ID is empty."
+case "$1" in
+  erase)
+    rm -f $SECRETS_FILE
+    CLEAN_REPO="$(git -C /etc/bridgehead remote get-url origin | sed -E 's|https://[^@]+@|https://|')"
+    git -C /etc/bridgehead remote set-url origin $CLEAN_REPO
+    exit 0
+    ;;
+  get)
+    # continue below
+    ;;
+  store)
+    # We could store the credentials in /var/cache/bridgehead, but we already did -- so nothing to do
+    exit 0
+    ;;
+  *)
+    fail_and_report 1 "gitpassword.sh called incorrectly"
+    ;;
+esac
 
 PARAMS="$(cat)"
 GITHOST=$(echo "$PARAMS" | grep "^host=" | sed 's/host=\(.*\)/\1/g')
 
-fetchVarsFromVault GIT_PASSWORD
+if [ ! -f ${SECRETS_FILE} ]; then
+   TMPFILE=$(mktemp)
+   docker run --rm \
+        -v $TMPFILE:/usr/local/cache \
+        -v $PRIVATEKEYFILENAME:/run/secrets/privkey.pem:ro \
+        -v /srv/docker/bridgehead/$PROJECT/root.crt.pem:/run/secrets/root.crt.pem:ro \
+        -v /etc/bridgehead/trusted-ca-certs:/conf/trusted-ca-certs:ro \
+        -e TLS_CA_CERTIFICATES_DIR=/conf/trusted-ca-certs \
+        -e NO_PROXY=localhost,127.0.0.1 \
+        -e ALL_PROXY=$HTTPS_PROXY_FULL_URL \
+        -e PROXY_ID=$PROXY_ID \
+        -e BROKER_URL=$BROKER_URL \
+        -e OIDC_PROVIDER=secret-sync-central.oidc-client-enrollment.$BROKER_ID \
+        -e SECRET_DEFINITIONS=GitLabProjectAccessToken:GIT_CONFIG_REPO_TOKEN:bridgehead-configuration \
```

Comment:
Also this is not how it is expected to be called I think it needs the full path to the repo which is of course a security concern as anyone with a valid beam cert could create a token for any other repo. Of course the disadvantage of auto generating the repo name from the beam task is that it might not match for some repos someone would have to check that but I guess its the better option

Reply:
Yes, we discussed that this morning and decided to change the syntax of `SECRET_DEFINITIONS` to this. The idea is to derive the proper url of the sites configuration repository based on broker id and site id.